packages/integrations/web-vitals/src/index.ts

```diff
@@ -0,0 +1,41 @@
+import { defineDbIntegration } from '@astrojs/db/utils';
+import { AstroError } from 'astro/errors';
+
+export default function webVitals() {
+	return defineDbIntegration({
+		name: '@astrojs/web-vitals',
+		hooks: {
+			'astro:db:setup'({ extendDb }) {
+				extendDb({ configEntrypoint: '@astrojs/web-vitals/db-config' });
+			},
+
+			'astro:config:setup'({ addMiddleware, config, injectRoute, injectScript }) {
+				if (!config.integrations.find(({ name }) => name === 'astro:db')) {
+					throw new AstroError(
+						'Astro DB integration not found.',
+						'Run `npx astro add db` to install `@astrojs/db` and add it to your Astro config.'
+					);
+				}
+
+				if (config.output !== 'hybrid' && config.output !== 'server') {
+					throw new AstroError(
+						'No SSR adapter found.',
+						'`@astrojs/web-vitals` requires your site to be built with `hybrid` or `server` output.\n' +
+							'Please add an SSR adapter: https://docs.astro.build/en/guides/server-side-rendering/'
+					);
+				}
+
+				// Middleware that adds a `<meta>` tag to each page.
+				addMiddleware({ entrypoint: '@astrojs/web-vitals/middleware', order: 'post' });
+				// Endpoint that collects metrics and inserts them in Astro DB.
+				injectRoute({
+					entrypoint: '@astrojs/web-vitals/endpoint',
+					pattern: '/_/astro-vitals',
```

Comment:
Currently injecting the data collection endpoint to `/_/astro-vitals/`.

Should we discuss whether we want a default namespacing approach? For example, the image service uses `/_image`.

Reply:
`/_/` sounds nice, I suppose we can only change `/_image` to it in a major since it's somewhat exposed?

Also bikeshedding 😅 Should we drop the `astro` and do `/_/web-vitals` or `/_/vitals` instead?

Reply:
Yeah, there are a bunch of potential approaches. I guess most extreme would be vendor-style namespacing matching the package name:

```
/_/astrojs/web-vitals/
```

Or alternatively, we don’t worry too much, and eventually support users customising the endpoint location to handle any conflicts.

Reply:
Naming is always tough! Maybe we could match the namespacing expected by `astro add`:
- use vendor prefixing for third party packages
- drop the "astro" for first party packages

So use either `/_/web-vitals/` or `/_web-vitals/`. I bias towards the latter to match `_images` and avoid introducing a new competing standard.

Reply:
Thanks for the feedback! Renamed to `/_web-vitals` and moved to a shared constant, so we can easily rename in the future if we want.